rfc/graph-feature-flags.md

```diff
@@ -0,0 +1,242 @@
+---
+title: "Graph Feature Flags: RFC to add feature flags into your Federated GraphQL API"
+author: Jens Neuse
+---
+
+This RFC proposes a new feature for Open Federation, which allows you to add feature flags to your Federated GraphQL API.
+This feature is called Graph Feature Flags, or GFF for short.
+
+GFF allows you to publish Subgraph Schema Changes under a feature flag.
+The runtime can dynamically enable or disable feature flags for each request.
+
+## Motivation
+
+There are four main reasons, and the correlated problems to solve, why we want to add Graph Feature Flags to Open Federation:
+
+1. **Feature Rollout**: Feature flags allow you to roll out new features gradually.
```

Comment:
Config synchronization is a topic here. Currently, every router operates in a standalone mode. As a User, I'd like to know if my roll-out was successfully. According to the consistency mode, we might need to reconsiders other strategies to keep it in sync and to report status updates in realtime.